docs/utils/useJsonColor.ts

```diff
@@ -56,36 +50,54 @@ export const useJsonColor = (): CompiledColors => {
       return value; // Return the direct value if not a reference.
     };
 
-    /**
-     * Compiles color themes from provided JSON theme definitions.
-     * Each color value is checked and potentially resolved using parseColorValue.
-     *
-     * @param themes - Object containing various theme definitions.
-     * @returns Object containing compiled and resolved themes.
-     */
     const compileColors = (themes: {
       [key: string]: Theme;
     }): CompiledColors => {
       const compiledColors: CompiledColors = {};
       Object.entries(themes).forEach(([themeName, theme]) => {
         const tempThemeColors: Theme = {};
         Object.entries(theme).forEach(([colorName, colorValues]) => {
-          const tempThemeColorPalette: ColorPalette = {};
-          Object.entries(colorValues).forEach(([shade, details]) => {
-            const { value, description } = details;
-            const resolvedValue = parseColorValue(value, figmaBrandColors);
-            const tempShadeColor = {
-              ...details,
-              value: resolvedValue,
-              description:
-                description + (value === resolvedValue ? '' : ` ${value}`),
+          compiledColors[themeName][colorName] = {};
+          if (colorValues.value) {
+            compiledColors[themeName][colorName] = {
+              ...colorValues,
+              value: colorValues.value,
+              description: colorValues.description,
             };
-            tempThemeColorPalette[shade] = tempShadeColor;
-          });
-          tempThemeColors[colorName] = tempThemeColorPalette;
+          } else {
+            Object.entries(colorValues).forEach(([shade, details]) => {
+              const { value, description } = details;
+              let resolvedValue = parseColorValue(value, figmaBrandColors);
+              if (!isHexColor(resolvedValue)) {
+                const cleanResolvedValue = resolvedValue
+                  .slice(1, -1)
+                  .split('.'); // Split the reference into parts
+                const category = cleanResolvedValue[0]; // Get the category (e.g., 'text')
+                const key = cleanResolvedValue[1]; // Get the key (e.g., 'default')
+
+                if (theme[category]?.[key]) {
+                  resolvedValue = parseColorValue(
+                    theme[category][key].value,
+                    brandColors,
+                  );
+                } else {
+                  console.error('Invalid reference:', resolvedValue);
+                }
+              }
+
+              compiledColors[themeName][colorName][shade] = {
+                ...details,
+                value: resolvedValue,
+                description:
+                  (description ?? '') +
+                  (value === resolvedValue ? '' : ` (${value})`),
+              };
+            });
+          }
         });
         compiledColors[themeName] = tempThemeColors;
       });
+
       return compiledColors;
```

Comment:
Will this file conflict with the updates here https://github.com/MetaMask/design-tokens/pull/699

Reply:
unfortunately yes and this one is most up to date because it takes into account the white and black colors @georgewrmarshall